### PR TITLE
Keep API Backward Compatibility with Express Gateway

### PR DESCRIPTION
**Pull request by brunokrebs**

### CODE REVIEW

File: _posts/2018-07-03-keep-api-backward-compatibility-with-express-gateway.markdown [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+It turns out that an [API Gateway][express-gateway] can solve this problem for you. Being an intermediate layer between the public interface **AND** the internal services, an Express Gateway can manipulate the requests and the responses during their journey.
+
+Here's what you would need to do to set up a gateway:
+
+- You would need to configure the [`proxy`][proxy] policy so that the request to the old endpoint get routed to the new, correct internal services.
+- You would need to configure the [`bodyModifier`][bodyModifier] policy to modify the request's body so that the new customer service can consume it. In the same way, you would use the same policy this time to modify the response object to make sure the `createdBy` property is correctly added.
+
+### Installing Express Gateway
+
+To solve these problems with Express Gateway, you would install and use a beta version of the framework. This version will most likely land in the `master`branch (thus as an official version) soon (the team supporting Express Gateway is still in the process of discussing some internal requirements, so they weren't able to make it happen officially yet).
```
Comment (XVincentX) [inline _posts/2018-07-03-keep-api-backward-compatibility-with-express-gateway.markdown]: > `master`branch

Put a space.

File: _posts/2018-07-03-keep-api-backward-compatibility-with-express-gateway.markdown [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+While these changes most likely make sense for your internal clients (due to business requirements for example), your existing third-party clients do not care that much about this change. So, while willing to migrate at a certain point in the future, flipping directly the switch would not be an option.
+
+### Keeping the Legacy API Intact Using Express Gateway
+
+It turns out that an [API Gateway][express-gateway] can solve this problem for you. Being an intermediate layer between the public interface **AND** the internal services, an Express Gateway can manipulate the requests and the responses during their journey.
+
+Here's what you would need to do to set up a gateway:
+
+- You would need to configure the [`proxy`][proxy] policy so that the request to the old endpoint get routed to the new, correct internal services.
+- You would need to configure the [`bodyModifier`][bodyModifier] policy to modify the request's body so that the new customer service can consume it. In the same way, you would use the same policy this time to modify the response object to make sure the `createdBy` property is correctly added.
```
Comment (XVincentX) [inline _posts/2018-07-03-keep-api-backward-compatibility-with-express-gateway.markdown]: Let's remove the `bodyModifier` link for now, as long we do not publish the thing in the `master` branch we won't have any official documentation.

**Comment by XVincentX** [comment]
Thanks for the corrections!

File: _posts/2018-07-03-keep-api-backward-compatibility-with-express-gateway.markdown [hunk 3]
```diff
@@ -0,0 +1,186 @@
+---
+layout: post
+title: "Keep API Backward Compatibility with Express Gateway"
+description: "Express Gateway is an API gateway that sits at the heart of any microservices architecture and that can help you keep your APIs backward compatible."
```
Comment (scottmathson-auth0) [inline _posts/2018-07-03-keep-api-backward-compatibility-with-express-gateway.markdown]: For the sake of avoiding duplicate content with other websites, can this be re-written before publication? Working "Express.js" keyword (high search vol.) into the meta description as well - thanks!

express-gateway-duplicate-meta

**Comment by XVincentX** [comment]
I can't modify this PR — somebody else will need to do that.

**Pull request merged**